core-processor/src/main/java/io/micronaut/expressions/context/ExpressionEvaluationContext.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.expressions.context;
+
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.inject.ast.MethodElement;
+import io.micronaut.inject.ast.TypedElement;
+
+import java.util.List;
+
+/**
+ * The context against which expressions are evaluated.
+ * Context methods, properties and method parameters can be referenced
+ * in evaluated expressions.
+ *
+ * @author Sergey Gavrilov
+ * @since 4.0.0
+ */
+@Internal
+public sealed interface ExpressionEvaluationContext permits BeanContextExpressionEvaluationContext,
```

Comment:
I've refactored this set of classes and separated compilation context from evalutaion context (which is used at runtime for evaluation process). Hopefully, now it will be more clear and easier to understand